pkg/apis/multiclusterdns/v1alpha1/federationdomain_types.go

```diff
@@ -0,0 +1,37 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// FederationDomain
+// +k8s:openapi-gen=true
+// +kubebuilder:resource:path=federationdomains
+type FederationDomain struct {
```

Comment:
Does it make sense to name this type `Domain` since it's part of the `multiclusterdns.federation.k8s.io` API group?

Reply:
I am fine. @pmorie wdyt?